src/askvm/resources/core/assign.ts

```diff
@@ -0,0 +1,55 @@
+import {
+  any,
+  resource,
+  run,
+  runUntyped,
+  Options,
+  TypedValue,
+  JSONable,
+} from '../../lib';
+
+function assignValue(
+  valueObject: { [key: string]: any },
+  key: string,
+  assignedValue: any
+) {
+  if (Object.isFrozen(valueObject[key]))
+    throw new Error(`Cannot assign to a constant variable "${key}"`);
+  valueObject[key] = assignedValue;
+}
+
+export const assignRes = resource({
+  type: any,
+  async compute(options, code, args): Promise<TypedValue<JSONable>> {
+    const { params: children = [] } = code;
+
+    const key: any = await runUntyped(options, children[0]); // FIXME any
+    const value = await run(options, children[1]);
+
+    if (key === 'resources') {
```

Comment:
This branch is not tested - please add a test for it.
I think actually this condition repeats itself in all of the relevant resources, right?
Might consider extracting it and making it more robust.

Reply:
Implemented test in `assign.test.ts`
Can I leave the extraction of this check to a separate module for a separate Github issue?

Reply:
Yea. Sure. Mind creating an issue for that?

Reply:
Aye, created #530 using Feature request template 😄